app/controllers/agendas.js

```diff
@@ -196,38 +225,48 @@ export default class AgendasController extends Controller {
 
     // default secretary
     const meetingSecretary = await newMeeting.secretary;
-    const currentApplicationSecretary = await this.mandatees.getCurrentApplicationSecretary();
-    const decisionActivity = this.store.createRecord(
-      'decision-activity',
-      {
-        startDate: startDate,
-        decisionResultCode,
-        secretary: meetingSecretary ? meetingSecretary : currentApplicationSecretary
-        // no subcase. Minutes approval aren't part of a (sub)case
-      }
-    );
+    const currentApplicationSecretary =
+      await this.mandatees.getCurrentApplicationSecretary();
+    const decisionActivity = this.store.createRecord('decision-activity', {
+      startDate: startDate,
+      decisionResultCode,
+      ...(this.enableDigitalAgenda
+        ? {
+            secretary: meetingSecretary
+              ? meetingSecretary
+              : currentApplicationSecretary,
+          }
+        : {}),
+      // no subcase. Minutes approval aren't part of a (sub)case
+    });
```

Comment:
Cleaning this up a little bit so the two awaits don't happen when the digital agenda flag is off